packages/api/tests/lambdas/test-sf-semaphore-down.js

```diff
@@ -170,6 +210,34 @@ test('sfSemaphoreDown lambda throws error for invalid event message', async (t)
   );
 });
 
+test('sfSemaphoreDown lambda decrements semaphore for s3-stored event message', async (t) => {
+  const status = 'SUCCEEDED';
+  const { client, semaphore } = t.context;
+  const queueName = randomId('low');
+  // arbitrarily set semaphore so it can be decremented
+  await client.put({
+    TableName: process.env.SemaphoresTable,
+    Item: {
+      key: queueName,
+      semvalue: 1
+    }
+  }).promise();
+
+  const stubReturn = createExecutionMessage(queueName);
+  const pullStepFunctionStub = sinon.stub(commonAws, 'pullStepFunctionEvent');
+  const proxiedFunction = proxyquire('../../lambdas/sf-semaphore-down', { pullStepFunctionEvent: pullStepFunctionStub }).handleSemaphoreDecrementTask;
+  pullStepFunctionStub.returns(stubReturn);
+  await proxiedFunction(
+    createCloudwatchPackagedEventMessage({ status, queueName })
+  );
+
+  const response = await semaphore.get(queueName);
+  t.is(response.semvalue, 0);
+
+  pullStepFunctionStub.restore;
```

Comment:
I think you need `pullStepFunctionStub.restore()`. Also, would it need to be in a try/finally in case `proxiedFunction` throws?

Reply:
Indeed.  Thanks for catching it.